src/MergeContext.js

```diff
@@ -108,6 +108,96 @@ class Approval {
     }
 }
 
+class StatusCheck
+{
+    constructor(context, state, targetUrl, description) {
+        assert(context);
+        assert(state);
+        assert(targetUrl);
+        assert(description);
+
+        this.context = context;
+        this.state = state;
+        this.targetUrl = targetUrl;
+        this.description = description;
+    }
+}
+
+// Passed status checks analysis
+class StatusResult
+{
+    // checksNumber:
+    //   for staged commits: the bot-configured number of required checks (Config.stagingChecks()),
+    //   for PR commits: GitHub configured number of required checks (requested from GitHub)
+    // context: either "PR" or "Staging";
+    constructor(checksNumber, context) {
+        assert(checksNumber !== undefined && checksNumber !== null);
+        assert(context);
+        this.checksNumber = checksNumber;
+        this.context = context;
+        this.statusChecks = [];
+    }
+
+    addStatus(statusCheck) {
```

Comment:
Please rename to addRequiredStatus to emphasize that the caller must filter out the tests that may fail.